src/pages/events/journaling-tutorial/runtime-action-code-journaling-api.md

```diff
@@ -0,0 +1,432 @@
+}
+ 
+/**
+ * Main entrypoint for the Adobe I/O action.
+ *
+ * @param {Object} params - Action input parameters.
+ * @param {string} params.LOG_LEVEL - (Optional) Log level (info, debug, error).
+ * @returns {Promise<Object>} - Returns a success response with fetched events or an error response.
+ */
+async function main(params) {
+  const logger = Core.Logger('main', { level: params.LOG_LEVEL || 'info' })
+  logger.info('Fetching events from journalling endpoint...')
+ 
+  try {
+    // Retrieve access token
+    const accessToken = await getAccessToken(params)
+    if (!accessToken) {
+      throw new Error('Missing access_token in params')
+    }
+ 
+    // Initialize state store (Adobe Runtime State)
+    const state = await stateLib.init()
+ 
+    // Load last saved journal position to continue from
+    const lastPositionRes = await state.get('last_journal_position')
+    const lastPosition = lastPositionRes ? lastPositionRes.value : undefined
+    logger.info(`Last Position >>> ${lastPosition}`)
+ 
+    // Fetch events from Adobe Journalling API
+    const events = await fetchEvent(params, accessToken, lastPosition)
+    logger.info(`Fetched ${events.length} new event(s).`)
+ 
+    if (events.length > 0) {
+      events.forEach((event, index) => {
+        logger.info(`--- Event #${index + 1} ---`)
+        logger.info(`Raw Payload:\n${JSON.stringify(event, null, 2)}`)
+ 
+        // Extract and log event details if available
+        if (event.event) {
+          logger.info(`Event ID: ${event.event.id}`)
+          logger.info(`Type: ${event.event.type}`)
+          logger.info(`Time: ${event.event.time}`)
+ 
+          if (event.event.data && event.event.data.value) {
+            const data = event.event.data.value
+            if (data.sku) logger.info(`SKU: ${data.sku}`)
+            if (data.name) logger.info(`Name: ${data.name}`)
+            if (data.price) logger.info(`Price: ${data.price}`)
+            if (data.quantity_and_stock_status?.qty !== undefined) {
+              logger.info(`Quantity in stock: ${data.quantity_and_stock_status.qty}`)
+            }
+          }
+        }
+      })
+ 
+      // Save newest journal position for next invocation
+      const newestPosition = events[events.length - 1].position
+      await state.put('last_journal_position', newestPosition) // default TTL
+      logger.info(`Updated last processed position: ${newestPosition}`)
+    }
+ 
+    // Return success response
+    return { success: true, events_fetched: events.length, events }
+  } catch (error) {
+    logger.error('Fetch failed with error:', error)
+    return errorResponse(500, error.message, logger)
+  }
+}
+ 
+exports.main = main
+```
+
+## Reading Through Adobe I/O Journals
+
+Adobe I/O Journaling API acts as a queue that holds events temporarily. Each event has a position marker that serves as a cursor. The fetchEvent() function retrieves events from the journaling URL provided in the parameters.
```

Comment:
Spelling Consistency: both `Journalling` and `Journaling` are used interchangeably in this pull request. I will recommend reviewing these instances and stick to using only one of them.